src/test/ui/stability-attribute/missing-const-stability.rs

```diff
@@ -1,12 +1,24 @@
 #![feature(staged_api)]
+#![stable(feature = "stable", since = "1.0.0")]
 
-#![stable(feature = "rust1", since = "1.0.0")]
+#[stable(feature = "stable", since = "1.0.0")]
+pub const fn foo() {} //~ ERROR function has missing const stability attribute
 
-#[stable(feature = "foo", since = "1.0.0")]
-pub const fn foo() {}
-//~^ ERROR rustc_const_stable
+#[unstable(feature = "unstable", issue = "none")]
+pub const fn bar() {} // ok for now
 
-#[unstable(feature = "bar", issue = "none")]
-pub const fn bar() {} // ok
+#[stable(feature = "stable", since = "1.0.0")]
+pub struct Foo;
+impl Foo {
+    #[stable(feature = "stable", since = "1.0.0")]
+    pub const fn foo() {} //~ ERROR associated function has missing const stability attribute
+
+    #[unstable(feature = "unstable", issue = "none")]
+    pub const fn bar() {} // ok for now
+}
+
+// FIXME When #![feature(const_trait_impl)] is stabilized, add tests for const
+// trait impls. Right now, a "trait methods cannot be stable const fn" error is
+// emitted, but that's not in the scope of this test.
```

Comment:
Might still be good to have that in this test, to ensure this test will be updated when that feature becomes stable.

Reply:
Will do.

Reply:
The order in which checks are performed seems to prevent this for the time being. If I add code intended to check this, _only_ the "trait methods cannot be stable const fn" is emitted; the errors introduced in this PR are not emitted, even for the unrelated lines above.

Reply:
Ah, I see, makes sense... not much we can do then I guess.

Reply:
I'll at least document that, given that I probably did the exact same thing I tried earlier before. Prevent others from wasting time attempting it again.